scripts/run_soupX.R

```diff
@@ -34,31 +37,139 @@ option_list <- list(
     opt_str = "--sample_name",
     type = "character",
     help = "Name associated with this sample"
+  ),
+  make_option(
+    opt_str = "--results_dir",
+    type = "character",
+    help = "output dir to put results",
+    default="./"
   )
 )
 
 #parse options
 opts <- parse_args(OptionParser(option_list = option_list))
 raw <- opts$raw
-fil <- opts$fil
+filtered <- opts$filtered
 clusters_file <- opts$cluster
 sample_name <- opts$sample_name
+results_dir <- opts$results_dir
+dir.create(results_dir)
 
 # load 10X data and convert to Soup Channel (object SoupX uses for analysis)
-fil_mat <- Read10X_h5(fil, use.names = T)
-raw_mat <- Read10X_h5(raw, use.names = T)
-clusters <- read.csv(clusters_file)
-mDat <- data.frame(clusters=clusters$Cluster,row.names=clusters$Barcode)
-sc <- SoupChannel(raw_mat, fil_mat, mDat)
+filtered_matrix <- Read10X_h5(filtered, use.names = TRUE)
+raw_matrix <- Read10X_h5(raw, use.names = TRUE)
+
+############################################################################
+# SoupX will use default pdf settings, set output file name to be more useful
+pdf(file = file.path(results_dir, paste0(sample_name, ".soupx.plots.pdf")))
+
+# Make a Seurat object from the sparce matrix
+seurat_obj  <- CreateSeuratObject(counts = filtered_matrix)
+
+# If cluster file available
+if (!is.null(clusters_file)) {
+  print("Clusters file exists!")
+  clusters <- read.csv(clusters_file)
+  mDat <- data.frame(clusters=clusters$Cluster, row.names=clusters$Barcode)
+  sc <- SoupChannel(raw_matrix, filtered_matrix, mDat)
+} else {
+  print("Clusters file does not exist. Let's calculate clusters by using Seurat.")  
+  # Make a “SoupChannel”, the object needed to run SoupX
+  sc  <- SoupChannel(raw_matrix, filtered_matrix)
+  print(sc)
```

Comment:
I see print statements in the Rscript. Is it left intentionally? Would it be better in the long run to capture the output/prints in a file. Maybe, create a log file

Reply:
Yeah, I agree. a log file is a good idea. I'll tackle that after lunch